vassal-app/src/main/java/VASSAL/counters/KeyBuffer.java

```diff
@@ -109,6 +121,18 @@ public Command keyCommand(javax.swing.KeyStroke stroke) {
     for (GamePiece p : targets) {
       bounds.addPiece(p);
       p.setProperty(Properties.SNAPSHOT, ((PropertyExporter) p).getProperties()); // save state prior to command
+      
+      // Send most recent click point location
+      GamePiece bp = Decorator.getInnermost(p);
+      if (bp instanceof BasicPiece) {
+        Command c3 = new NullCommand();
+        c3 = ((BasicPiece)bp).setPersistentProperty(BasicPiece.CLICKED_X, String.valueOf(clickPoint.x));
+        c3.append (((BasicPiece)bp).setPersistentProperty(BasicPiece.CLICKED_Y, String.valueOf(clickPoint.y)));
+        if ((comm != null) && !comm.isNull()) {
+          comm = comm.append(c3);
+        }
+      }              
```

Comment:
This entire If statement can be replaced by

comm = comm.append(setPersistentProperty(BasicPiece.CLICKED_X, String.valueOf(clickPoint.x));
comm = comm.append(setPersistentProperty(BasicPiece.CLICKED_Y, String.valueOf(clickPoint.y));

comm is already set to non-null and Decorator.setPersistentProperty will do all the work for you.

Reply:
I would chain it:

```
comm = comm
  .append(...)
  .append(...);
```

Reply:
That's what I went in with at your original suggestion but I'm getting `The method setPersistentProperty(String, String) is not defined for the type KeyBuffer`.   I looked for example code but surprisingly couldn't find any.

Reply:
I could do this...? That compiles.

`
      GamePiece bp = Decorator.getInnermost(p);
      if (bp instanceof BasicPiece) {
        comm = comm.append(((BasicPiece)bp).setPersistentProperty(BasicPiece.CLICKED_X, String.valueOf(clickPoint.x)))
                   .append(((BasicPiece)bp).setPersistentProperty(BasicPiece.CLICKED_Y, String.valueOf(clickPoint.y)));
      }              
`

Reply:
Looks sloppy in the stupid Github format, but looks right in code. Let me know if I'm somehow not finding a more direct method? Could that be something in a PR that's not merged yet?

Reply:
Oops, bad advice sorry.

    for (GamePiece p : targets) {
      bounds.addPiece(p);
      p.setProperty(Properties.SNAPSHOT, ((PropertyExporter) p).getProperties()); // save state prior to command
      if (p instanceof PersistentPropertyContainer) {
        comm = comm.append(((PersistentPropertyContainer) p).setPersistentProperty(BasicPiece.CLICKED_X, String.valueOf(clickPoint.x)))
          .append((PersistentPropertyContainer) p).setPersistentProperty(BasicPiece.CLICKED_Y, String.valueOf(clickPoint.y)));
      }
      comm = comm.append(p.keyEvent(stroke));
      bounds.addPiece(p);
    }

Reply:
Okay, should be all set now. 

BTW you are the King of mismatched parentheses in code fragments :D I think you might currently be 0-for-4 on those :D :D

Reply:
The in-built Java compiler in my head is a bit sloppy :)